### PR TITLE
Enable unicode support for watch

### PR DESCRIPTION
**Pull request by ThoFrank**

### CODE REVIEW

File: Formula/watch.rb [hunk 1]
```diff
@@ -25,9 +26,11 @@ class Watch < Formula
 
   def install
     system "autoreconf", "-fiv"
+    system "echo", "\"%s/\\\\(CURSES_LIB=ncurses\\\\)w/\\\\1/\\nwq\"", "|", "ed -s configure"
```
Comment (fxcoudert) [inline Formula/watch.rb]: Why do we need to hack the configure script?

Comment (yumitsu) [reply Formula/watch.rb]: @ThoFrank Why do you use `ed`, when you can `inreplace` without spawning another process? Not to mention, you're trying to solve the problem in a very strange way.

**Comment by yumitsu** [comment]
Take a look [here](https://github.com/yumitsu/homebrew-misc/blob/master/watch.rb) - I once did a formula for `watch`, but don't remember, why it didn't go.

File: Formula/watch.rb (hunk 1 above)
Comment (ThoFrank) [reply Formula/watch.rb]: Yeah I don't really know what I am doing there. Sorry for the mess.
The problem is that building watch with the "--enable-watch8bit" flag requires ncursesw. It exists on my machine, but is only a link to ncurses.
I found the hack googling around, but if you have an idea how to deal with the problem properly please feel free to help.

**Comment by jonchang** [comment]
macOS comes with `ncurses`, so to use the brewed version of `ncurses` you'll need to pass its location to `configure`.

**Comment by yumitsu** [comment]
@jonchang As far as I can remember, there is no such need as to pass LD/CPPFLAGS - if you added formula as build dependency to another, `brew` will install and set up buildenv by himself.

Also, about `macOS comes with`... Yes, it does, and that's why we are using `brew`-powered tools, like `bash`, `curl`, `wget`... 🤣

File: Formula/watch.rb (hunk 1 above)
Comment (yumitsu) [reply Formula/watch.rb]: @ThoFrank I will look into the matter today. As I remember, `watch` asked for `ncurses/ncursesw` only for `--enable-watch8bit`, but I don't think it really does matter if it 'classic' or 'narrow'.

**Comment by jonchang** [comment]
> Also, about _macOS comes with_... Yes, it does, and that's why we are using `brew`-powered tools, like `bash`, `curl`, `wget`... 🤣

My goal was not to point out that `ncurses` is provided by macOS, but rather specific instructions on how to link against brewed copies of the same. Given that this formula is currently failing to link against `ncurses`, it's pretty relevant to point it out.

**Comment by yumitsu** [comment]
@jonchang Well, that's why we had (and still do) `homebrew/dupes` - most libraries in macOS became obsolete a lot of time ago, not to mention that a lot of macOS libraries are not even POSIX/BDS compliant (e.g., `ncurses`, and it's a well-known fact).

**Comment by fxcoudert** [comment]
Closing for now, we would welcome an updated PR once it passes locally for you.

**Pull request closed without merging**